File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        start,end=0,len(arr)-1
        while start<=end:
            mid=start+(end-start)//2
            correctNum=mid+1
            missing=arr[mid]-correctNum
            if missing>=k:
                end=mid-1
            else:
                start=mid+1
        return start+k               

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution {
public:
    bool check(int missingCount,int k){
        if(missingCount>=k)return true;
        else return false;
    }
    int findKthPositive(vector<int>& arr, int k) {
        int n=arr.size();
        int low=0,high=n-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            int missingCount=arr[mid]-(mid+1);
            if(check(missingCount,k))high=mid-1;
            else low=mid+1;
        }
        return low+k;
    }
};